researchstack-mpower/src/main/java/org/sagebionetworks/research/mpower/researchstack/framework/MpTaskHelper.java

```diff
@@ -0,0 +1,231 @@
+ * other materials provided with the distribution.
+ *
+ * 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission. No license is granted to the trademarks of
+ * the copyright holders even if such marks are included in this software.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.sagebionetworks.research.mpower.researchstack.framework;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+
+import org.joda.time.DateTime;
+import org.researchstack.backbone.AppPrefs;
+import org.researchstack.backbone.ResourceManager;
+import org.researchstack.backbone.answerformat.AnswerFormat;
+import org.researchstack.backbone.model.TaskModel;
+import org.researchstack.backbone.result.FileResult;
+import org.researchstack.backbone.result.Result;
+import org.researchstack.backbone.result.StepResult;
+import org.researchstack.backbone.storage.NotificationHelper;
+import org.researchstack.backbone.task.Task;
+import org.sagebionetworks.bridge.android.manager.BridgeManagerProvider;
+import org.sagebionetworks.bridge.data.Archive;
+import org.sagebionetworks.bridge.data.ArchiveFile;
+import org.sagebionetworks.bridge.data.JsonArchiveFile;
+import org.sagebionetworks.bridge.researchstack.TaskHelper;
+import org.sagebionetworks.bridge.researchstack.factory.ArchiveFileFactory;
+import org.sagebionetworks.bridge.researchstack.survey.SurveyAnswer;
+import org.sagebionetworks.bridge.researchstack.wrapper.StorageAccessWrapper;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.MpSmartSurveyTask;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.body.MpBooleanAnswerFormat;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.body.MpChoiceAnswerFormat;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.body.MpIntegerAnswerFormat;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.body.MpTextQuestionBody;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MpTaskHelper extends TaskHelper {
```

Comment:
This should probably be renamed something like AnswerFileTaskHelper and put in backbone, but I'm trying to get this done ASAP.

Reply:
Please, lets keep this out of ResearchStack. It's *very* specific to how Bridge schemas are set up. While, ideally, we could move away from using the filename in our schema tables, that's not a project for today.

Reply:
I actually think TaskHelper only exists in Sage android-sdk, so this would just go in that same sage-specific library.